var/spack/repos/builtin/packages/py-sh/package.py

```diff
@@ -0,0 +1,37 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class PySh(PythonPackage):
+    """Python subprocess interface"""
+
+    homepage = "https://github.com/amoffat/sh"
+    url      = "https://pypi.io/packages/source/s/sh/sh-1.12.9.tar.gz"
+
+    version('1.12.9', 'ddc128a8d943d25afa6e01af11e0063b')
+    version('1.11',   '7af8df6c92d29ff927b6db0146bddec3')
+
+    depends_on('py-setuptools', type='build', when="@1.12.9:")
```

Comment:
These version constraints aren't right. I just checked `1.12.0` and it does require `setuptools`. I would either use `when='@1.12.0:'`, or leave out the version constraint altogether.